File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        lookup = {")": "(", "]": "[", "}": "{"}
        stack = []
        
        for param in s:
            if param in "({[":
                stack.append(param)
            else:
                if stack and stack[-1] == lookup[param]:
                    stack.pop()
                else:
                    return False
        
        if stack: return False
        return True

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    var lookup = {")": "(", "]": "[", "}": "{"};
    var stack = [];
    const open_paren = "([{"
    
    for(let i=0; i<s.length; i++){
        if(open_paren.includes(s[i])){
            stack.push(s[i]);
        }
        else{
            if(stack.length != 0 && stack[stack.length - 1] == lookup[s[i]]){
                stack.pop();
            }
            else{
                return false;
            }
        }
    }
    
    if(stack.length==0){
        return true;
    }
    return false;
    
};